### PR TITLE
WstxEventFactory: Ensure events' immutable non-null location

### PR DESCRIPTION
**Pull request by stanio**
Addresses #204:

-   Ensure events with non-null location
-   Ensure events with non-volatile location

The latter is a secondary issue noted in the original report.

### CODE REVIEW

File: src/main/java/com/ctc/wstx/stax/WstxEventFactory.java [hunk 1]
```diff
@@ -45,6 +48,28 @@ public WstxEventFactory() {
     /////////////////////////////////////////////////////////////
      */
 
+    @Override
+    public void setLocation(Location location) {
+        super.setLocation(location == null ? WstxInputLocation.getEmptyLocation()
```
Comment (stanio) [inline src/main/java/com/ctc/wstx/stax/WstxEventFactory.java:53]: As noted in the test:

https://github.com/FasterXML/woodstox/blob/2f1ed918bda9567a87f99cb89990907aa96114b0/src/test/java/wstxtest/evt/TestEventFactoryLocation.java#L58-L61

This returns a Location with an empty-string plublicId/systemId.  I could create a different constant in this class if deemed necessary:

```java
    private static WstxInputLocation UNKNOWN_LOCATION =
            new WstxInputLocation(null, null, (String) null, 0L, 0, 0);
```

**Comment by stanio** [comment]
@cowtowncoder, would you review this, when you get the time?

**Comment by cowtowncoder** [comment]
Hi @stanio! Thank you for contributing this & apologies for slow follow-up.

I am traveling for work this week but will try to review this when I get home.

**Comment by cowtowncoder** [comment]
@stanio I am now reviewing this and it looks good so far so I think there won't be issues.

But one last process thing before I can merge the fix is to get a CLA from

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

(unless you have sent one for Woodstox or Jackson contributions).
It's usually easiest to print, fill & sign, scan/photo, email to `cla` at fasterxml dot com.
Once we get it, CLA is good for any and all contributions to FasterXML projects so it's one-time-only thing.

Looking forward to merging this; will now complete the review.

File: src/main/java/com/ctc/wstx/stax/WstxEventFactory.java (hunk 1 above)
Comment (cowtowncoder) [reply src/main/java/com/ctc/wstx/stax/WstxEventFactory.java:53]: I think this is ok for now.

**Review by cowtowncoder — APPROVED** [review]
LGTM

**Comment by cowtowncoder** [comment]
Ok, so after getting CLA & merging, I think I'll backport this into 6.6 for inclusion in 6.6.3 (should be easy to cherry-pick).

Thank you again for contributing this!

**Comment by stanio** [comment]
I have sent my CLA earlier today.

**Comment by cowtowncoder** [comment]
Merged into `master` for eventual 7.0.0, but also backported into 6.x for 6.7.0

**Pull request merged**